socketio/src/client/raw_client.rs

```diff
@@ -288,8 +288,29 @@ impl RawClient {
                 }
 
                 if let Some(ref attachments) = socket_packet.attachments {
-                    if let Some(payload) = attachments.get(0) {
-                        ack.callback.deref_mut()(Payload::Binary(payload.to_owned()), self.clone());
+                    if let Some(payload) = attachments.first() {
+                        ack.callback.deref_mut()(
+                            Payload::Binary(payload.to_owned()),
+                            self.clone(),
+                        );
+                    }
+                    if ack.time_started.elapsed() < ack.timeout {
+                        if let Some(ref payload) = socket_packet.data {
+                            ack.callback.deref_mut()(
+                                Payload::from(payload.to_owned()),
+                                self.clone(),
+                            );
+                        }
+                        if let Some(ref attachments) = socket_packet.attachments {
+                            if let Some(payload) = attachments.first() {
+                                ack.callback.deref_mut()(
+                                    Payload::Binary(payload.to_owned()),
+                                    self.clone(),
+                                );
+                            }
+                        }
```

Comment:
Your are calling the callbacks three times in a row, is that expected behaviour? Wouldn't it be expected to call it once, but with all data at once?

Reply:
I'm not sure about this changes, I copy them from the OLD PR
maybe just remove them?

Reply:
Well, you are proposing a change here, so I would expect you could take a look at this to make it work?

https://github.com/socketio/socket.io-client should have it somewhere in its JS code :)

Reply:
https://github.com/socketio/socket.io-client/blob/main/lib/socket.ts#L467-L495
so, yea, just nothing to do, remove the ack id and ignore it